### PR TITLE
[MultiDomainBundle] Add MultiDomainBundle

### PR DESCRIPTION
**Pull request by wimvds**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | yes
| BC breaks?    | yes
| Deprecations? | no
| Fixed tickets | 

This PR contains some small bugfixes (ie. for search & adminlists) and includes the MultiDomainBundle that adds multi-domain & multi-site support.

### CODE REVIEW

File: src/Kunstmaan/AdminBundle/Helper/AdminPanel/DefaultAdminPanelAdaptor.php [hunk 1]
```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace Kunstmaan\AdminBundle\Helper\AdminPanel;
+
+use Symfony\Component\Security\Core\SecurityContextInterface;
+
+class DefaultAdminPanelAdaptor implements AdminPanelAdaptorInterface
```
Comment (krispypen) [inline src/Kunstmaan/AdminBundle/Helper/AdminPanel/DefaultAdminPanelAdaptor.php]: Maybe you should split this up in 3 different adaptors?

File: src/Kunstmaan/MultiDomainBundle/Controller/SiteSwitchController.php [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        }
+
+        $domainConfiguration = $this->get('kunstmaan_node.domain_configuration');
+        $host = $request->query->get('host');
+        $hosts = $domainConfiguration->getHosts();
+        if (!in_array($host, $hosts)) {
+            throw $this->createNotFoundException('Invalid host specified');
+        }
+
+        $this->get('session')->set(DomainConfiguration::OVERRIDE_HOST, $host);
```
Comment (krispypen) [inline src/Kunstmaan/MultiDomainBundle/Controller/SiteSwitchController.php]: Maybe show a big warning in the backend that you are showing another domain?, with a return button to make the override empty again?

File: src/Kunstmaan/MultiDomainBundle/Resources/config/routing.yml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+kunstmaanmultidomainbundle_site_switch:
+    resource: @KunstmaanMultiDomainBundle/Controller/SiteSwitchController.php
+    type:     annotation
+    prefix:   /admin/menu/
```
Comment (krispypen) [inline src/Kunstmaan/MultiDomainBundle/Resources/config/routing.yml]: is "menu" correct here?

File: src/Kunstmaan/NodeBundle/Controller/NodeAdminController.php [hunk 4]
```diff
@@ -120,29 +138,65 @@ public function copyFromOtherLanguageAction($id)
...
-        $otherLanguageNodeTranslation = $node->getNodeTranslation($otherlanguage, true);
-        $otherLanguageNodeNodeVersion = $otherLanguageNodeTranslation->getPublicNodeVersion();
-        $otherLanguagePage = $otherLanguageNodeNodeVersion->getRef($this->em);
-        $myLanguagePage = $this->get('kunstmaan_admin.clone.helper')->deepCloneAndSave($otherLanguagePage);
+        $otherLanguageNodeTranslation = $node->getNodeTranslation(
+            $otherlanguage,
+            true
+        );
+        $otherLanguageNodeNodeVersion = $otherLanguageNodeTranslation->getPublicNodeVersion(
+        );
```
Comment (krispypen) [inline src/Kunstmaan/NodeBundle/Controller/NodeAdminController.php]: strange code formatting?

File: src/Kunstmaan/MultiDomainBundle/Controller/SiteSwitchController.php (hunk 2 above)
Comment (wimvds) [inline src/Kunstmaan/MultiDomainBundle/Controller/SiteSwitchController.php]: A flash message will be shown in the admin if a host override is in effect.

File: src/Kunstmaan/AdminBundle/Helper/AdminPanel/DefaultAdminPanelAdaptor.php (hunk 1 above)
Comment (wimvds) [inline src/Kunstmaan/AdminBundle/Helper/AdminPanel/DefaultAdminPanelAdaptor.php]: That would make overriding the default behavior a lot harder... Right now you just have to override ```kunstmaan_admin.admin_panel.adaptor``` with your own custom implementation to change default behavior (you can inherit the current behavior from it as all methods are protected or public).

**Comment by krispypen** [comment]
@roderik :+1:

**Comment by yoshz** [comment]
Hi @wimvds,

I used the MultiDomainBundle to successfully implement multilingual site with a different domain for each locale.
However I'm not sure how I can easily generate an url to a translation on a different hostname.
This functionality seems te be missing in DomainBasedLocaleRouter.

In pull request #570 I was achieving this by setting the host parameter when you generate url.

Any thoughts about a proper solution?

**Pull request merged**